solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -2133,18 +2132,18 @@ public static AddUpdateCommand convertTlogEntryToAddUpdateCommand(SolrQueryReque
       new SolrNamedThreadFactory("recoveryExecutor"));
 
 
-  public static void deleteFile(File file) {
+  public static void deleteFile(Path file) {
     boolean success = false;
     try {
-      Files.deleteIfExists(file.toPath());
+      Files.deleteIfExists(file);
       success = true;
     } catch (Exception e) {
       log.error("Error deleting file: {}", file, e);
     }
 
     if (!success) {
       try {
-        file.deleteOnExit();
+        file.toFile().deleteOnExit();
```

Comment:
Just looks a bit weird with `file.toFile()`, but probably ok. Have not checked if there is a deleteOnExit API on Files..

Reply:
There's not a deleteOnExit API for NIO. The options are to either convert to a File or register our own shutdown hooks (which is what this does behind the scenes). I think this code exists specifically for Windows running with an AV (or the test FileSystem that simulates this) so I'm not sure what the other replacement options are.